ott/core/initializers.py

```diff
@@ -278,6 +282,252 @@ def init_dual_a(
     return f_u
 
 
+@jax.tree_util.register_pytree_node_class
+class MetaOTInitializer(DefaultInitializer):
+  """Meta OT Initializer :cite:`amos:22`.
+
+  This initializer predicts the solution to the coupling problem
```

Comment:
I think we should insist a bit more on the fact that this is an initializer in the "fixed" geometry setting. As a result, maybe think about naming this initializer to something that introduces that distinction now (rather than later, if we expect initializers that learn potential functions), e.g. FixedGeometryMetaOTInitializer (maybe too long :) ). I gave it a try:


This initializer consists in a predictive model that outputs, given input probability weights `a` and `b`, two initialization vectors (i.e. approximate solutions) `f,g` to solve their (dual) Entropy regularized OT problem. The model's parameters are learned using a training set of OT instances (multiple pairs of probability weights), assuming the **same** geometry is used throughout. The meta model defaults to the MLP in
  :class:`~ott.core.initializers.Meta_MLP` and, with batched problem instances passed into ``update``.

Reply:
Yeah, I like that. Updated in https://github.com/ott-jax/ott/pull/145/commits/4f5d3035b500aad6b49f6db4e2136cd72b2d1a3f

An alternative would be to support fixed and non-fixed geometries in the same interface, e.g. by allowing an optional argument `fixed_geom` that is set when the user wants a fixed geometry but otherwise indicates that the OT problems have individual geometries. This may make the interface a little cleaner, but it may complicate how the model should be specified (e.g. with a fixed geometry we can assume the meta model just takes `a,b` as inputs rather than also depending on the geometry. Because of this, I like separating the fixed geometry case (as the code is doing now) more than trying to have the right interface around both cases.

There's also the question of whether `Meta_MLP` should be called `FixedGeometry_Meta_MLP`. I'd be okay keeping it shorter as it's unlikely that an MLP would be the ideal model when varying the geometry, but we could also update it now too. What do you think @marcocuturi?

Reply:
i think it's ok indeed! this is great!! I also think we might think of refactoring something later in a single `MetaOTInitializer` class, but for now this is cool!